web/server/vue-cli/src/components/Product/ProductConfigForm.vue

```diff
@@ -228,10 +266,23 @@ export default {
         this.productConfig.displayedName_b64 =
           value.length ? window.btoa(value) : null;
       }
-    }
+    },
   },
 
+  created () {
+    if (!this.productConfig.confidentiality) {
+      this.productConfig.confidentiality = Confidentiality.CONFIDENTIAL;
+    }
+  },
   methods: {
+    encodeValue(confidentialityId) {
+      return this.confidentialityFromCodeToString(confidentialityId);
+    },
+
+    decodeValue(confidentialityName) {
+      return this.confidentialityFromStringToCode(confidentialityName);
+    },
```

Comment:
THere is no need to declare these functions. Use the encode/decode functions explicitly.